packages/react-core/src/components/Menu/MenuListItem.tsx

```diff
@@ -0,0 +1,179 @@
+  /** Content rendered inside the menu list item. */
+  children?: React.ReactNode;
+  /** Additional classes added to the menu list item */
+  className?: string;
+  /** Identifies the component in the Menu onSelect or onFavorite callback */
+  itemId?: any;
+  /** Target navigation link */
+  to?: string;
+  /** Flag indicating whether the item is active */
+  isActive?: boolean;
+  /** Callback for item click */
+  onClick?: (event: React.MouseEvent) => void;
+  /** Component used to render the menu item */
+  component?: React.ReactNode;
+  /** Render item as disabled option */
+  isDisabled?: boolean;
+  /** Render item with icon */
+  icon?: React.ReactNode;
+  /** Render item with action */
+  action?: React.ReactNode;
+  /** Description of the menu item */
+  description?: React.ReactNode;
+  /** Render external link icon */
+  isExternalLink?: boolean;
+  /** Internal flag indicating if the option is selected */
+  isSelected?: boolean;
+  /** Render expandable icon */
+  isExpandable?: boolean;
+  /** Flag indicating if the item is favorited */
+  isFavorite?: boolean;
+  /** Aria label text for favoritable button when favorited */
+  ariaIsFavoriteLabel?: string;
+  /** Aria label text for favoritable button when not favorited */
+  ariaIsNotFavoriteLabel?: string;
+  /** Flyout menu */
+  flyoutMenu?: React.ReactNode;
+  /** Callback function when mouse leaves trigger */
+  onShowFlyout?: (event?: MouseEvent) => void;
+  /** Accessibility label */
+  'aria-label'?: string;
+  /** Forwarded ref */
+  innerRef?: React.Ref<any>;
+}
+
+const MenuListItemBase: React.FunctionComponent<MenuListItemProps> = ({
+  children,
+  className,
+  itemId,
+  to,
+  isActive = false,
+  flyoutMenu,
+  description = null as string,
+  onClick = () => {},
+  component,
+  isDisabled = false,
+  isExternalLink = false,
+  isExpandable = false,
+  isSelected = false,
+  icon,
+  action,
+  isFavorite,
+  ariaIsFavoriteLabel,
+  ariaIsNotFavoriteLabel,
+  onShowFlyout,
+  innerRef,
+  ...props
+}: MenuListItemProps) => {
+  const Component = component || to ? 'a' : 'button';
+  const [flyoutVisible, setFlyoutVisible] = React.useState(false);
+
+  const showFlyout = (displayFlyout: boolean) => {
+    setFlyoutVisible(displayFlyout);
+    onShowFlyout && displayFlyout && onShowFlyout();
+  };
+
+  const onItemSelect = (event: React.MouseEvent, onSelect: any) => {
+    // Trigger callback for Menu onSelect
+    onSelect(event, itemId);
+    // Trigger callback for item onClick
+    onClick(event);
+  };
+
+  const renderDefaultLink = (onSelect: any): React.ReactNode => {
+    const additionalProps =
+      Component === 'a'
+        ? {
+            href: to
+          }
+        : {};
+    return (
+      <>
+        <Component
+          onClick={(event: React.MouseEvent) => onItemSelect(event, onSelect)}
+          className={css(
+            styles.menuItem,
+            isFavorite && styles.modifiers.favorited,
```

Comment:
This item shouldn't need `pf-m-favorited`. Only the `menuItemAction` element below that has `.pf-m-favorite` would toggle `.pf-m-favorited` when it's favorited. You can see that in the core examples - https://pf4.patternfly.org/components/menu#with-favorites